### PR TITLE
Refactor tasks on first-login

### PR DESCRIPTION
**Pull request by sharat87**
Primarily doing example app cloning and
analytics in parallel.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/authentication/handlers/AuthenticationSuccessHandler.java [hunk 1]
```diff
@@ -51,10 +51,12 @@
                 : handleRedirect(webFilterExchange);
 
         return sessionUserService.getCurrentUser()
-                .flatMap(user -> user.getExamplesOrganizationId() == null
-                        ? analyticsService.sendEvent(AnalyticsEvents.FIRST_LOGIN, user)
-                        : Mono.empty())
-                .then(examplesOrganizationCloner.cloneExamplesOrganization())
+                // TODO: Need a better way to identify if this is the user's first-login.
+                .filter(user -> user.getExamplesOrganizationId() == null)
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/authentication/handlers/AuthenticationSuccessHandler.java:55]: If this filter returns `false`, with the `then(redirectionMono)` still be invoked? Have we tested that bit?

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/authentication/handlers/AuthenticationSuccessHandler.java:55]: Yes, it goes through. The call `.then` will proceed on completion of previous mono, with data or empty.

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/authentication/handlers/AuthenticationSuccessHandler.java:55]: Oh, if there's an error then it won't proceed. Do we want the redirection to happen even if there's an error in analytics or cloning?

**Review by mohanarpit — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java [hunk 2]
```diff
@@ -71,11 +71,6 @@
      * @return Empty Mono.
      */
     private Mono<Organization> cloneExamplesOrganization(User user) {
-        if (user.getExamplesOrganizationId() != null) {
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java:74]: Can we keep this check? I know it's duplicated but it ensures that the example apps will not get created even by mistake.

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java:74]: Done

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**